core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -302,9 +373,12 @@ void sendBuddyConnectResponseMessage(UUID senderUserID, UUID senderUserAnonymize
 
 	private void disconnectBuddy(UserAnonymizedDTO userAnonymized, UUID userAnonymizedID)
```

Comment:
Consider to rename to `disconnectBuddyIfPresent`. By the way, how can it be that the `buddyAnonymized` is not present while the sending and receiving status is `ACCEPTED` (as tested in the caller of this method)?